template/apps/api/src/resources/user/user.service.ts

```diff
@@ -23,15 +23,55 @@ const updateLastRequest = (_id: string) => {
   );
 };
 
+const updateSubscription = async (data: any) => {
+  const subscription = {
+    subscriptionId: data.id,
+    priceId: data.plan.id,
+    productId: data.plan.product,
+    status: data.status,
+    interval: data.plan.interval,
+    currentPeriodStartDate: data.current_period_start,
+    currentPeriodEndDate: data.current_period_end,
+    cancelAtPeriodEnd: data.cancel_at_period_end,
+  };
+
+  service.atomic.updateOne(
+    {
+      stripeId: data.customer,
+    },
+    {
+      $set: {
+        subscription,
+      },
+    },
+  );
+};
+
+const deleteSubscription = async (data: any) => {
+  service.atomic.updateOne(
+    {
+      stripeId: data.customer,
+    },
+    {
+      $unset: {
+        subscription: '',
+      },
+    },
+  );
+};
+
 const privateFields = [
   'passwordHash',
   'signupToken',
   'resetPasswordToken',
+  'subscription',
```

Comment:
I don't think subscription is a private field. And we don't need creating separate endpoint to retrieve it.